fdbserver/storageserver.actor.cpp

```diff
@@ -644,6 +644,91 @@ struct BusiestWriteTagContext {
 	    busiestWriteTagEventHolder(makeReference<EventCacheHolder>(busiestWriteTagTrackingKey)), lastUpdateTime(-1) {}
 };
 
+// A SSPhysicalShard represents a physical shard, it contains a list of keyranges.
+class SSPhysicalShard {
+public:
+	SSPhysicalShard() : id(0LL) {}
+	SSPhysicalShard(const int64_t id) : id(id) {}
+
+	void addRange(Reference<ShardInfo> shard);
+	void removeRange(Reference<ShardInfo> shard);
+	void removeRange(KeyRangeRef range);
```

Comment:
These two `removeRange` need comments. From what the code does, they are different: The first one seems to remove only if the `shard` pointers are the same; the second one remove any overlapping `shard`s.

Reply:
Added comments.